dds_web/api/dds_decorators.py

```diff
@@ -31,6 +40,99 @@
 # S3 ########################################################################################## S3 #
 
 
+def project_optional(func):
+    """Verify that user has access to project."""
+
+    @functools.wraps(func)
+    def verify_access(*args, **kwargs):
+        """Check for access."""
+
+        project_info = flask.request.args
+        if project_info and project_info.get("project"):
+            project = project_schemas.ProjectRequiredSchema().load(project_info)
+        else:
+            project = None
+
+        return func(*args, project=project, **kwargs)
+
+    return verify_access
+
+
+def renew_access_required(func):
+    """Check that user has permission to give access to another user in this project."""
+
+    @functools.wraps(func)
+    def access_decorator(*args, user, project, **kwargs):
+        """Check if the current user has access to renew project access."""
+        if auth.current_user() == user:
+            raise AccessDeniedError(message="You cannot renew your own access.")
+
+        # Get roles
+        current_user_role = get_user_roles_common(user=auth.current_user())
+        other_user_role = get_user_roles_common(user=user)
+
+        # Check if Researcher and if so is project owner or not
+        if other_user_role == "Researcher" and project:
+            project_user_row = models.ProjectUsers.query.filter_by(
+                project_id=project.id, user_id=user.username
+            ).one_or_none()
+            if project_user_row and project_user_row.owner:
+                other_user_role = "Project Owner"
```

Comment:
I think this is taken care of by the `get_user_roles_common` no? Since `project` should still be in `flask.request.arg`?